### PR TITLE
Fix build for GHC-9.0

### PR DESCRIPTION
**Pull request by tvh**

### CODE REVIEW

File: src/Network/HTTP2/Client.hs
Comment (ProofOfKeags) [inline src/Network/HTTP2/Client.hs:286]: Why was this done? I'd imagine this hurts GHC's ability to do inlining and if you are explicitly wanting to stop inlining, the pragma would be a better way to accomplish that.

Comment (tvh) [reply src/Network/HTTP2/Client.hs:286]: See this bit of the release notes:

> GHC now implements simplified subsumption, as described in GHC Proposal #287. This change simplifies the type system, and prevents the possibility of GHC silently changing the semantics of user programs, but it does mean that some libraries may need eta-expansion to typecheck. More info here: Subsumption.

**Review by ProofOfKeags — APPROVED** [review]
ACK 7366f97a7cd6d0f72a2c46d067b7ccd8d49d6676

I have not pulled and tested the code yet. It obviously looks semantically invariant, but I'm presently unsure about lifting the upper bound on `http2` since there are namespace collisions with the module naming last I remember. I will try to pull this down and use it in something later today. If someone else gets to it before me please document your findings here.

**Comment by ProofOfKeags** [comment]
Actually, as I said in the earlier comment, this will not work.

You can see here: https://hackage.haskell.org/package/http2-3.0.2

http2 has a module called `Network.HTTP2.Client` which interferes with the `Network.HTTP2.Client` exported by this library. As such they will never really be compatible. I'm actually somewhat surprised that your tests showed that this would work. Have you tried pulling in `http2-client` into another project with these changes?

**Comment by tvh** [comment]
Yes, but I am also using `PackageImports`. See https://github.com/memrange/push-notify-apn/pull/13/commits/c93cacb2d1f45201fa5655ff2ccc7789d5460bf5.

File: http2-client.cabal
Comment (ProofOfKeags) [inline http2-client.cabal:30]: There will be module namespace conflicts here. The correct way to support GHC-9 and http2-3.x is to rework the module naming scheme for this library, I think. While I'm aware that package namespacing exists, requiring users of this library to use that in order to use this library may be more onerous than necessary. This is especially true since some of the functionality is being replicated by both libraries.

**Review by ProofOfKeags — CHANGES_REQUESTED** [review]

File: http2-client.cabal
Comment (tvh) [reply http2-client.cabal:30]: Yes, I agree that using `PackageImports` may be the bigger evil here.

**Comment by ProofOfKeags** [comment]
> Yes, but I am also using PackageImports. See memrange/push-notify-apn@c93cacb.

Making this change will essentially turn -XPackageImports into a required extension for use with this lib. I am not tremendously upset about that. But I'm also only a recent maintainer of the package. If I can get a signoff from @lucasdicioccio or examples of other widely used libraries that have bit the bullet with this particular issue, then I'm happy to merge.

**Comment by ethercrow** [comment]
I use this library and am completely fine with embracing `-XPackageImports`.

**Comment by lucasdicioccio** [comment]
Closing as we seem to have a working 9.x build now (e.g., 9.8 with #92), thanks for all involved and apologies for the glacially-slow time to update.

**Pull request closed without merging**
